src/http/server.cr

```diff
@@ -131,6 +131,11 @@ class HTTP::Server
   # Creates a `TCPServer` listenting on `host:port` and adds it as a socket, returning the local address
   # and port the server listens on.
   #
+  # ```
+  # server = HTTP::Server.new { }
```

Comment:
More related to the implementation than this doc, but initializing with an empty Proc looks a bit weird.

Reply:
Agreed, but these examples are to demonstrate the bind methods, not handlers. So I think this is fine.
We could add some hello world to the handler, but this just adds house not relevant to the  immediate purpose.